### PR TITLE
Modernize for deployment via actions

### PR DESCRIPTION
**Pull request by babbitt**
TL;DR: I noticed that deployments have been few and far between likely due to version incompatibility and the use of Travis. I created a workflow that deploys prose to gh-pages using [github actions](https://github.com/features/actions) which will get new versions deployed consistently.

Longer story:
I recently came upon prose and I think it's a great project. I noticed that the docs didn't align with the [live site](https://prose.io/) since the last deploy hasn't happened in over 6 years. After spending sometime getting my own fork running I wanted to share the love and try contributing back to the main project. 

**Changes in summary**
I needed to start by just getting the codebase to build on a modern version of node (I'm using v20 LTS) <br>
Most notable among these is gulp (migrating from v3 -> v4) due to a shim in [graceful-fs failing](https://stackoverflow.com/a/58394828). This necessitated re-writing the `gulpfile` to make it v4 compatible. I haven't used the tool before and am mostly hacking around here, so it works, but from my understanding is more of a port than a proper re-write with best practices syntax.  

The other notable changes to the `package.json` come in the form of version bumps. These do not appear to cause any breaking changes in real-world testing as well as unit test runs to the best of my knowledge. Figured I'd start the modernization effort small.

Now that  I had prose building on my system I knew that github actions would be able to replicate the same and wrote a deploy.yml file that largely copies the functionality of the deploy.sh script that used to run. The only difference is that the built files get sent directly to a binary that is uploaded/deployed by gh-pages directly, removing the need for a gh-pages branch. 

**Testing**
I've given this build a good amount of real world testing via my own branch and the unit tests seem to come back looking good as well, that being said, I can't get a great "before" picture so take a grain of salt. In regards to testing a the github-actions file. I'm using an almost identical one for [my fork]( https://github.com/babbitt/prosa) but just to be sure I spun up a branch with these exact changes and it [built successfully](https://github.com/babbitt/prosa/actions/runs/7484205191)

As I said I'm not entirely familiar with the stack and code base so happy to go back and forth if there are styling issues or other work required for this change. Looking forward to hopefully getting some momentum building on this project again.

P.S. This should address #1136

### CODE REVIEW

**Comment by thiagohersan** [comment]
This is great. Good to see stuff happening in the repo and that people (other than me) use this tool.

Reading this summary now I remembered going down a rabbit hole of trying to update gulp and other prose dependencies, but it turned out to be more than a weekend-project for me, so I gave up.

All this to say "Big Ups" to @babbitt ! 🙃
